File: app/src/main/java/com/arctouch/codechallenge/api/interceptors/LanguageAndRegionInterceptor.java
package com.arctouch.codechallenge.api.interceptors;

import java.io.IOException;
import java.util.Locale;

import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class LanguageAndRegionInterceptor implements Interceptor{

    private static final String TAG = "LanguageInterceptor";

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request original = chain.request();
        HttpUrl originalHttpUrl = original.url();

        String language = Locale.getDefault().toString();
        String region = Locale.getDefault().getCountry();

        HttpUrl url = originalHttpUrl.newBuilder()
                .addQueryParameter("language", language)
                .addQueryParameter("region", region)
                .build();

        Request.Builder requestBuilder = original.newBuilder()
                .url(url);

        Request request = requestBuilder.build();
        return chain.proceed(request);
    }
}

File: app/src/main/java/com/arctouch/codechallenge/api/interceptors/LanguageAndRegionInterceptor.kt
package com.arctouch.codechallenge.api.interceptors

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException
import java.util.*

class LanguageAndRegionInterceptor : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val original = chain.request()
        val originalHttpUrl = original.url()

        val language = Locale.getDefault().toString()
        val region = Locale.getDefault().country

        val url = originalHttpUrl.newBuilder()
                .addQueryParameter("language", language)
                .addQueryParameter("region", region)
                .build()

        val requestBuilder = original.newBuilder()
                .url(url)

        val request = requestBuilder.build()
        return chain.proceed(request)
    }

    companion object {

        private val TAG = "LanguageInterceptor"
    }
}
